purchases/src/main/kotlin/com/revenuecat/purchases/virtualcurrencies/VirtualCurrencyManager.kt

```diff
@@ -85,16 +100,46 @@ internal class VirtualCurrencyManager(
             return null
         }
 
-        return deviceCache.getCachedVirtualCurrencies(appUserID = appUserID)
+        val cachedVirtualCurrencies: VirtualCurrencies? = deviceCache.getCachedVirtualCurrencies(appUserID = appUserID)
+        if (cachedVirtualCurrencies == null) {
+            log(LogIntent.DEBUG) {
+                VirtualCurrencyStrings.NO_CACHED_VIRTUAL_CURRENCIES
+            }
+        }
+
+        return cachedVirtualCurrencies
     }
 
-    @Suppress("UnusedParameter", "ForbiddenComment")
     private fun fetchVirtualCurrenciesFromBackend(
         appUserID: String,
         isAppBackgrounded: Boolean,
         callback: GetVirtualCurrenciesCallback,
-    ): VirtualCurrencies {
-        // TODO: Implement this
-        return VirtualCurrencies(all = emptyMap())
+    ) {
+        backend.getVirtualCurrencies(
+            appUserID = appUserID,
+            appInBackground = isAppBackgrounded,
+            onSuccess = { callback.onReceived(it) },
+            onError = { callback.onError(it) },
+        )
+    }
+
+    private fun fetchedVirtualCurrenciesFromBackendCallback(
+        originalCallback: GetVirtualCurrenciesCallback,
+        appUserID: String,
+    ): GetVirtualCurrenciesCallback = object : GetVirtualCurrenciesCallback {
+        override fun onReceived(virtualCurrencies: VirtualCurrencies) {
+            log(LogIntent.RC_SUCCESS) {
+                VirtualCurrencyStrings.VIRTUAL_CURRENCIES_UPDATED_FROM_NETWORK
+            }
+
+            cacheVirtualCurrencies(virtualCurrencies, appUserID)
+            originalCallback.onReceived(virtualCurrencies)
+        }
+        override fun onError(error: PurchasesError) {
+            log(LogIntent.RC_SUCCESS) {
+                VirtualCurrencyStrings.VIRTUAL_CURRENCIES_UPDATED_FROM_NETWORK_ERROR.format(error)
+            }
+            originalCallback.onError(error)
```

Comment:
Not saying we should, but I wonder: should we clear the cache at this point?
FWIW, following the code, this point would only be reached if the cache is stale